src/librustc_mir/shim.rs

```diff
@@ -842,29 +841,26 @@ fn build_call_shim<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
     mir
 }
 
-pub fn build_adt_ctor<'a, 'gcx, 'tcx>(infcx: &infer::InferCtxt<'a, 'gcx, 'tcx>,
-                                      ctor_id: hir::HirId,
-                                      fields: &[hir::StructField],
-                                      span: Span)
-                                      -> Mir<'tcx>
-{
-    let tcx = infcx.tcx;
-    let gcx = tcx.global_tcx();
-    let def_id = tcx.hir().local_def_id_from_hir_id(ctor_id);
-    let param_env = gcx.param_env(def_id);
+pub fn build_adt_ctor<'a, 'gcx>(tcx: TyCtxt<'a, 'gcx, 'gcx>, ctor_id: DefId) -> &'gcx Mir<'gcx> {
```

Comment:
```suggestion
pub fn build_adt_ctor<'gcx>(tcx: TyCtxt<'_, 'gcx, 'gcx>, ctor_id: DefId) -> &'gcx Mir<'gcx> {
```

Reply:
If you're removing a lifetime parameter from the `<...>`, you can remove all of them anyway (since we have the relevant feature enabled for most `rustc` crates and we'll eventually fully transition to no explicit lifetime declarations).